warehouse/static/js/warehouse/utils/webauthn.js

```diff
@@ -30,18 +30,23 @@ const populateWebAuthnErrorList = (errors) => {
   });
 };
 
-const doWebAuthn = (buttonId, func) => {
+const doWebAuthn = (formId, func) => {
   if (!window.PublicKeyCredential) {
     return;
   }
 
-  const webAuthnButton = document.getElementById(buttonId);
-  if (webAuthnButton === null) {
+  const webAuthnForm = document.getElementById(formId);
+  if (webAuthnForm === null) {
     return null;
   }
 
+  const webAuthnButton = webAuthnForm.querySelector("button[type=submit]");
   webAuthnButton.disabled = false;
-  webAuthnButton.addEventListener("click", async () => { func(webAuthnButton.value); });
+
+  webAuthnForm.addEventListener("submit", async() => {
+    func(webAuthnButton.value);
```

Comment:
This should probably be:
```suggestion
    await func(webAuthnButton.value);
```

Reply:
Thanks @webknjaz.
If you want to make a new PR with this update, please do so :)

Reply:
Here you go: https://github.com/pypa/warehouse/pull/6608

Reply:
thanks!